### ISSUE TITLE
Cleanly use Eloquent alongside OAuthUserWrapper

### ISSUE THREAD

**Issue by obahareth** [kronthto/laravel-oauth2-login#5]
Hey,

So it'd be really cool if I could have the same kind of object returned from `Auth::user()` also inherit from `Eloquent\Model` (to be able to have a hasMany relation for example).

What's clean way to have my wrapper inherit from Eloquent\Model?

I can't do that at the moment because `AuthFromRequest` currently does

```php
$wrapperClass = config('oauth2login.authWrapper');
return new $wrapperClass($resourceOwner);
```

and this conflicts with the constructor signature of `Model` because it takes in an array rather than a `ResourceOwnerInterface`.

I'd like to be able to simply do `Auth::user()->some_has_many_relation` for example.

How do you suggest I accomplish this?

The simplest (and very hacky) way I found is to inherit from `Model` and make a constructor that takes a single parameter, if that parameter is an array (in which case my instance is initialized through Eloquent) simply call the parent constructor with it, if it gets a `ResourceOwnerInterface` (which I don't use that much because I actually use a JWT access token that has everything I need) instead then it's been initialized from laravel-oauth2-login. This method is quite brittle and I'd prefer to find a cleaner way.

I apologize again if I'm flooding you with questions, still finding my way through Laravel after years of not using it.

**Comment by kronthto** [comment]
Just for clarification, the `User` model in your consumer application can be filled from both, an `users` table and via OAuth2-Login? Designing the package I assumed that the user-data would always only be stored on the side of the IP.

However I acknowledge that even then you could have tables with a FK on the `user_id` and it would be nice to utilize relations for that.

I don't think there are many other solutions than the one you found, because of the design of this package. Hardwiring that the wrapper-class gets the `ResourceOwnerInterface` as param1 is bad style, this should be abstracted somehow.

My current approach for that would be to allow a factory/invokeable for the creation of the object to be specified in config and then: 
```php
return app(config('oauth2login.authWrapperFactory'))($resourceOwner);
```
In this factory you could obv. get the data as an assoc-array and construct/return the Eloquent model.

---

You mentioned you have relevant data in the JWT payload - have you managed to access that? Currently it (the raw AccessToken) is only saved in session I think. But probably decoding the JWT / injection the data to `Auth`/Request attributes should be done in your code. I don't think that concern fits in this package, as the AccessTokens are not necessarily JWTs.

**Comment by obahareth** [comment]
The factory approach seems like a good solution to the problem.

Yes I do use a JWT and I agree this library shouldn't concern itself with that, I got the JWT out of the session and decoded it myself.

**Comment by kronthto** [comment]
I just released `v1.6.0` with the added config option / code change. Does that solve your problem?

**Comment by obahareth** [comment]
Yes, this has solved all my issues! Thank you so much @kronthto!
